### ISSUE TITLE
Add command to tell the bot to delete one of its messages

### ISSUE THREAD

**Issue by benduhoux** [benduhoux/Sazerac#39]
# Feature description
Some messages are not ephemeral and could be send by mistake or need clean up.

# Proposed solution
Add a command to would take a message ID and the bot will delete it.

# Additional info
Message ID can be retrieve with right click if discord is in dev mode.
Might need to check that the bot is indeed the author to avoid deleting other messages in the case permissions were given for that.

**Comment by aminabrhm** [comment]
I hope it is an easy one, I would love to help!

**Comment by benduhoux** [comment]
It is a good first issue in the sense that the actual callback command is likely a one liner, with the addition of the message owner check maybe 3 lines of simple code. It would allow someone to focus on understanding how to add a command rather than implementing a more complex feature.

If you want to contribute, feel free to assign the issue to yourself and open a new branch. I'd likely advise to wait for #36 and #38  to be closed before doing this as the way commands are implemented will change.

If you don;t mind me asking, how did you find out about this repo ?

**Comment by benduhoux** [comment]
This might not be as easy as I initially thought.

The two delete methods are either from the `Interaction` object through `delete_original_response` OR from `InteractionMessage `using `delete`.

A simple right-click on a non ephemeral response from the bot returns what I assume is the `Message.id` corresponding to the `InteractionMessage `from the bot response.

From there, I suppose we could use the `delete `method but the problem I am running into is that I do not know how to recover the `Message `or `InteractionMessage `from the ID that I got from the right-click.

**Comment by benduhoux** [comment]
Turns out a few of the models used in the regular API have a `fetch_message()` method that returns the `Message `object with the desired ID (see https://discordpy.readthedocs.io/en/stable/api.html#messageable).
I assume they are not described explicitly in the Interaction API docs because the equivalent classes inherit from the regular API ones.

Either way, I now have a solution that I'll push.

I wanted this quickly-ish to clean-up some testing that ended up posting messages where it should not have so I grabbed it myself.
